src/user/PreferencesController.ts

```diff
@@ -182,10 +182,9 @@ export class PreferencesController extends BaseController<BaseConfig, Preference
 		addresses = addresses.map((address: string) => toChecksumAddress(address));
 		const oldIdentities = this.state.identities;
 		const identities = addresses.reduce((ids: { [address: string]: ContactEntry }, address, index) => {
-			ids[address] = {
+			ids[address] = oldIdentities[address] || {
```

Comment:
This change seems functionally equivalent, as long as the `name` and `address` properties are always set on pre-existing identities (which I think is true?). Not sure what it has to do with the other changes in this PR though.

Reply:
I had issues with typescript errors, without those changes I couldn't build it. About your question, yes they're always defined.

![image](https://user-images.githubusercontent.com/12115171/88595209-8cf38a80-d030-11ea-8d3b-22339676c39a.png)

Reply:
Ah I see - that makes sense. Thanks!

Reply:
This seems similar to #243—@estebanmino what version of TS are you using?

Reply:
3.8.3

Reply:
Is the same issue I had, but I'm on 3.8.3